pkg/plugin/plugin.go

```diff
@@ -0,0 +1,253 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"cmp"
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/lima-vm/lima/v2/pkg/usrlocalsharelima"
+)
+
+type Plugin struct {
+	Name        string `json:"name"`
+	Path        string `json:"path"`
+	Description string `json:"description,omitempty"`
+}
+
+func DiscoverPlugins() ([]Plugin, error) {
+	var plugins []Plugin
+	seen := make(map[string]bool)
+
+	dirs := getPluginDirectories()
+
+	for _, dir := range dirs {
+		pluginsInDir := scanDirectory(dir)
+		for _, plugin := range pluginsInDir {
```

Comment:
This may be a personal preference, but I prefer to avoid temp variables that are only used once, unless they are necessary to keep the expressions short. But I find this easier to read:

```suggestion
	for _, dir := range getPluginDirectories() {
		for _, plugin := range scanDirectory(dir) {
```

Of course this only works because the functions don't return an error.